cmd/crossplane/core/core.go

```diff
@@ -103,6 +103,7 @@ type startCommand struct {
 	TLSServerCertsDir   string `env:"TLS_SERVER_CERTS_DIR"   help:"The path of the folder which will store TLS server certificate of Crossplane."`
 	TLSClientSecretName string `env:"TLS_CLIENT_SECRET_NAME" help:"The name of the TLS Secret that will be store Crossplane's client certificate."`
 	TLSClientCertsDir   string `env:"TLS_CLIENT_CERTS_DIR"   help:"The path of the folder which will store TLS client certificate of Crossplane."`
+	TLSAllowInsecure    bool   `help:"Allow insecure TLS connections to Crossplane functions"`
```

Comment:
If we add an `--insecure` flag I think it should affect all TLS, not only functions.

We actually use these same certs to connect to external secret stores. See the `if c.EnableExternalSecretStores` block. I think it was an oversight that we load the same certs twice. I thought we used them to authenticate validation webhooks as well, but I can't see where that is plumbed up (CC @phisco who might know).

Reply:
Done. I implemented the flag for external secret stores as well. @negz @phisco